src/components/content/document/patient-documents.tsx

```diff
@@ -16,15 +17,20 @@ export type PatientDocumentProps = {
 function PatientDocumentsComponent({ className }: PatientDocumentProps) {
   const containerRef = useRef<HTMLDivElement>(null);
   const breakpoints = useBreakpoints(containerRef);
-  const patientDocumentQuery = usePatientDocument();
+  const query = usePatientDocument();
   const { featureFlags } = useCTW();
   const openDetails = useResourceDetailsDrawer({
     header: (m) => `${m.dateCreated} - ${m.title}`,
     details: documentData,
   });
 
-  const document = patientDocumentQuery.data ?? [];
-  const { isLoading } = patientDocumentQuery;
+  const document = query.data ?? [];
+
+  useEffect(() => {
+    if (!query.isLoading) {
+      Telemetry.reportZAPRecordCount("documents", query.data?.length);
+    }
+  }, [query.isLoading, query.data]);
```

Comment:
same here (checking query.data exists before calling `reportZAPRecordCount`
```suggestion
  useEffect(() => {
    if (!query.isLoading && query.data) {
      Telemetry.reportZAPRecordCount("documents", query.data.length);
    }
  }, [query.isLoading, query.data]);
```